data_structures/lowest_common_ancestor.py

```diff
@@ -0,0 +1,118 @@
+# Find the lowest common ancestor. Given a Binary tree and two nodes
+
+# Node for a binary tree
+class Node:
+    """Node of the binary tree."""
+
+    def __init__(self, data, left=None, right=None):
+        self.data = data
+        self.left = left
+        self.right = right
+
+    def __str__(self) -> str:
+        return "({})".format(self.data)
+
+
+class BinaryTree:
+    """Binary tress."""
+
+    def __init__(self, py_list=[]):
+
+        list_len = len(py_list)
+
+        if list_len is None:
+            print("Binary list empty")
+
+        self.root = None
+
+        for i in py_list:
+            self.add_node(i)
+
+    def add_node(self, data):
+
+        if self.root is None:
+            self.root = Node(data)
+        else:
+            self._add_helper(self.root, data)
+
+    def _add_helper(self, sub_root, data):
+
+        if data < sub_root.data:
+            if sub_root.left is None:
+                sub_root.left = Node(data)
+            else:
+                self._add_helper(sub_root.left, data)
+        else:
+            if sub_root.right is None:
+                sub_root.right = Node(data)
+            else:
+                self._add_helper(sub_root.right, data)
+
+    def __iter__(self):
+        def inorder_traversal(sub_root):
+
+            if sub_root.left:
+                yield from inorder_traversal(sub_root.left)
+
+            yield sub_root
+
+            if sub_root.right:
+                yield from inorder_traversal(sub_root.right)
+
+        yield from inorder_traversal(self.root)
+
+    def __str__(self) -> str:
+        out = []
+        for n in self:
+            out.append(str(n.data))
+
+        return ", ".join(out)
+
+
+def find_ancestor(root, n1, n2):
```

Comment:
The way this function is written makes no sense to me. I am not sure what it even does.